lib/Dialect/FIRRTL/PassDetails.h

```diff
@@ -0,0 +1,23 @@
+//===- PassDetails.h - FIRRTL pass class details ----------------*- C++ -*-===//
+//
+//===----------------------------------------------------------------------===//
+
+// clang-tidy seems to expect the absolute path in the header guard on some
+// systems, so just disable it.
+// NOLINTNEXTLINE(llvm-header-guard)
```

Comment:
I was seeing some issues related to what John mentioned before https://github.com/llvm/circt/pull/157#discussion_r506115553.

The convention for these header guards is to construct the define according to the relative path within the project. However, I was seeing clang-tidy failing in this file because it was using the absolute path to this file on the host machine, which didn't match. I'll try to see why that happens and how to work around it.

I have a feeling we'd have this issue in lib/Dialect/LLHD/PassDetails.h as well, but we only run clang-tidy on files that are changed, so it hasn't come up.

Reply:
I'm giving myself a TODO to look into this further. After John's [comment](https://github.com/llvm/circt/pull/229#issuecomment-725618324) about this on another PR, I think for now we should just disable the check.